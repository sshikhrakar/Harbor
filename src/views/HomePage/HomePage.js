import React from 'react';
import { View, Text } from 'react-native';

function HomePage() {

  return (
    <View>
      <Text> HomePage </Text>
    </View>
  );

}

HomePage.propTypes = {
};

export default HomePage;
