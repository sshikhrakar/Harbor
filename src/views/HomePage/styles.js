import { StyleSheet } from 'react-native';

import colors from '../../config/colors';

const styles = StyleSheet.create({
  contentContainerStyle: {
    flex: 1,
  },

  mainContainer: {
    flex: 1,
    padding: 10,
    backgroundColor: colors.SILVER,
  },
});

export default styles;
