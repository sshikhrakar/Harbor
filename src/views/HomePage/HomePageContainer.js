import React, { Component } from 'react';

import HomePage from './HomePage';
import fonts from '../../config/fonts';
import images from '../../config/images';
import colors from '../../config/colors';

class HomePageContainer extends Component {

  constructor(props) {
    super(props);

    props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  /**
   * Navigation event handler.
   *
   * @param {event} event
   */
  onNavigatorEvent(event) {
    if (event.type === 'NavBarButtonPress') {
      if (event.id === 'addProject') {
        console.log('Add button pressed.'); // eslint-disable-line no-console
      }
    }
  }

  render() {
    return (
      <HomePage { ...this.props } />
    );
  }

}

/*
 * Navigation bar style parameters.
 */
HomePageContainer.navigatorStyle = {
  navBarTextFontSize: 20,
  navBarTextFontFamily: fonts.primary.REGULAR,
  navBarComponentAlignment: 'center',
  navBarBackgroundColor: colors.SILVER,
};

HomePageContainer.navigatorButtons = {
  rightButtons: [
    {
      icon: images.plusSign,
      id: 'addProject',
      buttonColor: colors.BLACK,
      disableIconTint: true,
      buttonFontSize: 12,
    },
  ],
};


export default HomePageContainer;
