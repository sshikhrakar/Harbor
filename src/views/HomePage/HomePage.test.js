import React from 'react';
import renderer from 'react-test-renderer';

import HomePage from './index';

describe('HomePage View', () => {

  it('should render correctly', () => {
    const tree = renderer.create(
      <HomePage />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
