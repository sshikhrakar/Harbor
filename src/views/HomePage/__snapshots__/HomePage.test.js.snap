// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage View should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={undefined}
  refreshControl={
    <RefreshControlMock
      onRefresh={undefined}
      refreshing={undefined}
    />
  }
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ecf0f1",
          "flex": 1,
          "padding": 10,
        }
      }
    />
  </View>
</RCTScrollView>
`;
